File: golang/app/repositories/HeatRepo.go
package repositories

import (
	"app/app/models"
	"gopkg.in/mgo.v2/bson"
)

type HeatRepo struct {
	model models.Heat
}

func (u *HeatRepo) Init() {
	u.model = models.Heat{}
}

func (u *HeatRepo) GetAllUserHasHeatZone(filters bson.M) []models.Heat {
	models, _ := u.model.Get(filters)
	return models
}

func (u *HeatRepo) ShowAll(filters bson.M) []models.Heat {
	models, _ := u.model.Get(filters)
	return models
}

func (u *HeatRepo) Store(body bson.M) (bson.M, error) {

	heat := models.Heat{
		User: body["user"].(string),
		X: int(body["x"].(float64)),
		Y: int(body["y"].(float64)),
	}

	u.model.Create(&heat)

	return body, nil
}

File: php72/crud/app/Repositories/HeatRepo.php
<?php

namespace App\Repositories;

use Illuminate\Http\Response;
use App\Models\Heat;

/**
 * Class ApiController
 *
 * @package App\Http\Controllers
 * @version 1.0
 * @author Angelo (LED) <angelolandino@htomail.it>
 * @copyright Copyright (c) 2015-present
 */
class HeatRepo
{
    function __construct(Heat $heat)
    {
        $this->model = $heat;
    }

    public function getAll($filters = []): array
    {
        return $this->model->get($filters)->toArray();
    }

    public function whereRaw($filters = []): array
    {
        return $this->model->whereRaw($filters)->get()->toArray();
    }

    public function store(array $data): array
    {
        return $this->model->create($data)->toArray();
    }
}